src/services/tupleOf.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable @typescript-eslint/only-throw-error */
+import { Param, ParamGetSet } from '@/types/paramTypes'
+import { ExtractParamType } from '@/types/params'
+import { getParamValue, setParamValue } from '@/services/params'
+
+type TupleOf<T extends Param[]> = { [K in keyof T]: ExtractParamType<T[K]> }
+
+export function tupleOf<const T extends Param[]>(...params: T): ParamGetSet<TupleOf<T>> {
+  return {
+    get: (value) => {
+      const values = value.split(',')
+
+      return params.map((param, index) => getParamValue(values.at(index), param)) as TupleOf<T>
+    },
+    set: (value, { invalid }) => {
+      if (!Array.isArray(value)) {
+        throw invalid('Expected an array')
```

Comment:
```suggestion
        throw invalid('Expected a tuple')
```

Reply:
this isn't a meaningful difference at runtime, but the error might be easier to debug if its not the same as `arrayOf`